internal/cmd/fluxtest-harness-influxdb/test.go

```diff
@@ -0,0 +1,233 @@
+			Init: &ast.StringLiteral{Value: b.Name},
+		},
+	}
+	orgOpt := &ast.OptionStatement{
+		Assignment: &ast.VariableAssignment{
+			ID:   &ast.Identifier{Name: "org"},
+			Init: &ast.StringLiteral{Value: t.l.Org.Name},
+		},
+	}
+
+	// During the first execution, we are performing the writes
+	// that are in the testcase. We do not care about errors.
+	_ = t.executeWithOptions(bucketOpt, orgOpt, t.writeOptAST, pkg)
+
+	// Execute the read pass.
+	return t.executeWithOptions(bucketOpt, orgOpt, t.readOptAST, pkg)
+}
+
+func (t *testExecutor) executeWithOptions(bucketOpt, orgOpt *ast.OptionStatement, optionsAST *ast.File, pkg *ast.Package) error {
+	options := optionsAST.Copy().(*ast.File)
+	options.Body = append([]ast.Statement{bucketOpt, orgOpt}, options.Body...)
+
+	// Add options to pkg
+	pkg = pkg.Copy().(*ast.Package)
+	pkg.Files = append(pkg.Files, options)
+
+	// Use testing.inspect call to get all of diff, want, and got
+	inspectCalls := stdlib.TestingInspectCalls(pkg)
+	pkg.Files = append(pkg.Files, inspectCalls)
+
+	bs, err := json.Marshal(pkg)
+	if err != nil {
+		return err
+	}
+
+	req := &query.Request{
+		OrganizationID: t.l.Org.ID,
+		Compiler:       lang.ASTCompiler{AST: bs},
+	}
+
+	r, err := t.l.FluxQueryService().Query(t.ctx, req)
+	if err != nil {
+		return err
+	}
+	defer r.Release()
+
+	for r.More() {
+		v := r.Next()
+
+		if err := v.Tables().Do(func(tbl flux.Table) error {
+			// The data returned here is the result of `testing.diff`, so any result means that
+			// a comparison of two tables showed inequality. Capture that inequality as part of the error.
+			// XXX: rockstar (08 Dec 2020) - This could use some ergonomic work, as the diff testOutput
+			// is not exactly "human readable."
+			return fmt.Errorf("%s", table.Stringify(tbl))
+		}); err != nil {
+			return err
+		}
+	}
+	r.Release()
+	return r.Err()
+}
+
+// This options definition puts to() in the path of the CSV input. The tests
+// get run in this case and they would normally pass, if we checked the
+// results, but don't look at them.
+const writeOptSource = `
+import "testing"
+import c "csv"
+
+option testing.loadStorage = (csv) => {
+	return c.from(csv: csv) |> to(bucket: bucket, org: org)
+}
+`
+
+// This options definition is for the second run, the test run. It loads the
+// data from previously written bucket. We check the results after running this
+// second pass and report on them.
+const readOptSource = `
+import "testing"
+import c "csv"
+
+option testing.loadStorage = (csv) => {
+	return from(bucket: bucket)
+}
+`
+
+func prepareOptions(optionsSource string) *ast.File {
+	pkg := parser.ParseSource(optionsSource)
+	if ast.Check(pkg) > 0 {
+		panic(ast.GetError(pkg))
+	}
+	return pkg.Files[0]
+}
+
+func (t *testExecutor) Logf(s string, i ...interface{}) {
+	_, _ = fmt.Fprintf(&t.errOutput, s, i...)
+	_, _ = fmt.Fprintln(&t.errOutput)
+}
+
+func (t *testExecutor) Errorf(s string, i ...interface{}) {
+	t.Logf(s, i...)
```

Comment:
This interface, that _looks_ a lot like `testing.T` interface, is quite neat. I like it a lot, and will likely take out my hack in idpe and put this in its place. Thanks for the idea.